server/service/appconfig_test.go

```diff
@@ -382,3 +382,97 @@ func TestModifyAppConfigSMTPConfigured(t *testing.T) {
 	require.False(t, dsAppConfig.SMTPSettings.SMTPEnabled)
 	require.False(t, dsAppConfig.SMTPSettings.SMTPConfigured)
 }
+
+// TestTransparencyURL tests that Fleet Premium licensees can use custom transparency urls and Fleet
+// Free licensees are restricted to the default transparency url.
+func TestTransparencyURL(t *testing.T) {
+	ds := new(mock.Store)
+
+	admin := &fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)}
+	ctx := viewer.NewContext(context.Background(), viewer.Viewer{User: admin})
+
+	checkLicenseErr := func(t *testing.T, licenseTier string, shouldFail bool, err error) {
+		if shouldFail {
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "requires Fleet Premium license")
+		} else {
+			require.NoError(t, err)
+		}
+	}
+	testCases := []struct {
+		name             string
+		licenseTier      string
+		initialURL       string
+		newURL           string
+		expectedURL      string
+		shouldFailModify bool
+	}{
+		{
+			"customURL",
+			"free",
+			defaultTransparencyURL,
+			"customURL",
+			defaultTransparencyURL,
+			true,
+		},
+		{
+			"customURL",
+			"premium",
+			defaultTransparencyURL,
+			"customURL",
+			"customURL",
+			false,
```

Comment:
Shouldn't this fail because of the `url.Parse` check?

Reply:
TIL:
```
The url may be relative (a path, without a host) or absolute (starting with a scheme). Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities.
```

Reply:
Probably ok as-is for this iteration.

As we don't know if in the future we would want to support disabling the feature by making Fleet Server return the config as `""` for instance.